File: 1 basic/1014. 福尔摩斯的约会 (20).cpp
#include<iostream>
#include<string>
using namespace std;

int min(int a,int b){
	return a<b?a:b;
} 
char days[7][4]={"MON","TUE","WED","THU","FRI","SAT","SUN"};

bool isdigit(char ch){
	return '0'<=ch && ch<='9';
}
bool isalpha(char ch){
	return ('A'<= ch && ch<= 'Z' )|| ('a'<= ch && ch<='z');
}

int main(){
	string a,b,c,d;
	cin>>a>>b>>c>>d;
	int mark;
	//output the weekday
	for(int i = 0,len = min(a.size(),b.size()); i < len; ++i){
		if(a[i] == b[i] && 'A'<=a[i] && a[i]<='G'){
			cout<<days[a[i]-'A']<<" ";
			mark = i;
			break;
		}
	}
	//output hh
	for(int i=mark+1,len = min(a.size(),b.size()); i < len; ++i){
		if(a[i] == b[i]){
			if(isdigit(a[i])){
				int x = a[i] - '0';
				cout<<x/10<<x%10<<":";
				break;
			}else if('A'<= a[i] && a[i]<='N'){
				int x = a[i] - 'A' + 10;
				cout<<x/10<<x%10<<":";
				break;
			}
		}
	}
	//output mm
	for(int i=0,len = min(c.size(),d.size()); i<=len; ++i){
		if(c[i] == d[i] && isalpha(c[i])){
			cout<<i/10<<i%10<<endl;
			break;
		}
	}
	
	return 0; 
} 

/*
0A0B0C
1A1B1C
ABCDE
Abcde
MON 11:00


0G0N0C
1G1N1C
12345678912345678912123A
12345678912345678912123A
SUN 23:23

0A00A000a
1A11A111a 
01234567890123456789012345678901234567890123456789012345678a
01234567890123456789012345678901234567890123456789012345678a
MON 10:59

3485djDkxh4hhGE 
2984akDfkkkkggEdsb 
s&hgsfdk 
d&Hyscvnm
THU 14:04

*/

File: 1 basic/py/1014. 福尔摩斯的约会 (20).py
a = input()
b = input()
c = input()
d = input()
len1,len2 = min( len(a),len(b)),min(len(c ),len(d ))
wd = {'A':'MON','B':'TUE','C':'WED','D':'THU','E':'FRI','F':'SAT','G':'SUN'}
tm = {'0':'00','1':'01','2':'02','3':'03','4':'04','5':'05','6':'06','7':'07',
      '8':'08','9':'09','A':'10', 'B':'11','C':'12','D':'13','E':'14','F':'15',
      'G':'16','H':'17','I':'18','J':'19','K':'20','L':'21','M':'22','N':'23'}
mark ,ans = 0,''
for i in range(len1):
    if a[i] == b[i] and 'A' <= a[i] <= 'G':
       ans = ans + wd[a[i]] + ' '
       mark = i + 1
       break
for i in range(mark, len1):
    if a[i]== b[i] and ('A'<=a[i]<='N' or '0'<=a[i ]<='9'):
        ans = ans + tm[a[i ]]
        break
for i in range(len2):
    if c[i ]== d[i ]:
        if 'A'<=c[i]<='Z' or 'a'<=c[i]<='z' :
            ans = ans + ':{0:02d}'.format(i)
            break
print(ans)